### ISSUE TITLE
Pointcloud

### ISSUE THREAD

**Issue by bqdrat** [apple/ml-depth-pro#11]
Is there a way to create pointcloud using your model?

**Comment by bqdrat** [comment]
I made it using this code

`
from plyfile import PlyData, PlyElement

def get_pcd_base(H, W, u0, v0, fx, fy):
    x_row = np.arange(0, W)
    x = np.tile(x_row, (H, 1))
    x = x.astype(np.float32)
    u_m_u0 = x - u0

    y_col = np.arange(0, H)  # y_col = np.arange(0, height)
    y = np.tile(y_col, (W, 1)).T
    y = y.astype(np.float32)
    v_m_v0 = y - v0

    x = u_m_u0 / fx
    y = v_m_v0 / fy
    z = np.ones_like(x)
    pw = np.stack([x, y, z], axis=2)  # [h, w, c]
    return pw


def reconstruct_pcd(depth, fx, fy, u0, v0, pcd_base=None, mask=None):
    if type(depth) == torch.__name__:
        depth = depth.cpu().numpy().squeeze()
    depth = cv2.medianBlur(depth, 5)
    if pcd_base is None:
        H, W = depth.shape
        pcd_base = get_pcd_base(H, W, u0, v0, fx, fy)
    pcd = depth[:, :, None] * pcd_base
    if mask:
        pcd[mask] = 0
    return pcd

apple_pcd = reconstruct_pcd(depth, fx=focallength_px.detach().cpu().numpy(), fy=focallength_px.detach().cpu().numpy(), 
                               u0=width / 2, v0=height / 2)
`
